lib/ansible/modules/packaging/language/yarn.py

```diff
@@ -0,0 +1,335 @@
+    description:
+      - Use the --ignore-scripts flag when installing.
+    required: false
+    choices: [ "yes", "no" ]
+    default: no
+  production:
+    description:
+      - Install dependencies in production mode.
+      - Yarn will ignore any dependencies under devDependencies in package.json
+    required: false
+    choices: [ "yes", "no" ]
+    default: no
+  registry:
+    description:
+      - The registry to install modules from.
+    required: false
+  state:
+    description:
+      - Installation state of the named node.js library
+      - If absent is selected, a name option must be provided
+    required: false
+    default: present
+    choices: [ "present", "absent", "latest" ]
+requirements:
+    - Yarn installed in bin path (typically /usr/local/bin)
+'''
+
+EXAMPLES = '''
+- name: Install "imagemin" node.js package.
+  yarn:
+    name: imagemin
+    path: /app/location
+- name: Install "imagemin" node.js package on version 5.3.1
+  yarn:
+    name: imagemin
+    version: '5.3.1'
+    path: /app/location
+- name: Install "imagemin" node.js package globally.
+  yarn:
+    name: imagemin
+    global: yes
+- name: Remove the globally-installed package "imagemin".
+  yarn:
+    name: imagemin
+    global: yes
+    state: absent
+- name: Install "imagemin" node.js package from custom registry.
+  yarn:
+    name: imagemin
+    registry: 'http://registry.mysite.com'
+- name: Install packages based on package.json.
+  yarn:
+    path: /app/location
+- name: Update all packages in package.json to their latest version.
+  yarn:
+    path: /app/location
+    state: latest
+'''
+
+RETURN = '''
+changed:
+    description: Whether Yarn changed any package data
+    returned: always
+    type: boolean
+    sample: true
+msg:
+    description: Provides an error message if Yarn syntax was incorrect
+    returned: failure
+    type: string
+    sample: "Package must be explicitly named when uninstalling."
+invocation:
+    description: Parameters and values used during execution
+    returned: success
+    type: dictionary
+    sample: {
+            "module_args": {
+                "executable": null,
+                "globally": false,
+                "ignore_scripts": false,
+                "name": null,
+                "path": "/some/path/folder",
+                "production": false,
+                "registry": null,
+                "state": "present",
+                "version": null
+            }
+        }
+out:
+    description: Output generated from Yarn with emojis removed.
+    returned: always
+    type: string
+    sample: "yarn add v0.16.1[1/4] Resolving packages...[2/4] Fetching packages...[3/4] Linking dependencies...[4/4]
+    Building fresh packages...success Saved lockfile.success Saved 1 new dependency..left-pad@1.1.3 Done in 0.59s."
+'''
+
+import os
+import re
+
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    import json
+except ImportError:
+    try:
+        import simplejson as json
```

Comment:
json module is in stdlib since Python 2.6, which is lowest version supported: https://github.com/ansible/ansible/blob/devel/setup.py#L206

So there's no need to do fallback.

Reply:
Is this a blocker for PR approval?

Reply:
no, but it's nice to avoid having "dead code" :)